### PR TITLE
Remove client-side usage of moment.js

### PR DESCRIPTION
**Pull request by ebroder**
We used a very narrow subset of moment.js in client-side code which
probably doesn't justify the 60 KB or so that it adds to bundle size.
Instead, replace it with a hand-tooled function that replicates similar
formatting rules.

(Note that this isn't identical because I realized on inspection that
our old handling of dates in the past 7 days vs. the future 7 days was a
little ambiguous. I don't think this comes up super often since I don't
think we show a lot of future-dated things, but this seems less
ambiguous so a bit better.)

### CODE REVIEW

**Comment by ebroder** [comment]
I left the momentjs usage in the server-side code for the time being. It'd be pretty easy if we wanted to actually scrub the dependency entirely, but I don't feel like it's that worthwhile to be hyper-strict about server-side dependencies.

**Review by zarvox — APPROVED** [review]
It looks like there's only a couple of serverside usages, and they're both just subtracting some number of seconds from `new Date()`, so I'd be perfectly happy if we just went ahead and got rid of the serverside usages too, and switched to `luxon` or one of the other more modern datetime libraries if we need something like that again.

File: imports/lib/calendarTimeFormat.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  const diff = now.getTime() - d.getTime();
+  if (diff > 0 && diff < 6 * 24 * 60 * 60 * 1000) {
+    return thisWeekFormatter.format(d);
+  }
+
+  return defaultFormatter.format(d);
+}
+
+// Extra compact formatting used in the chat window on the puzzle page
+calendarTimeFormat.short = (d: Date, now: Date = new Date()): string => {
```
[the remark this answers is not in the document]
Comment (ebroder) [reply imports/lib/calendarTimeFormat.ts]: Yeah fine. I felt a little gross about it at the time, so I'll switch to that. PTAL?

**Comment by ebroder** [comment]
Reworked to drop the server dependency

**Review by zarvox — APPROVED** [review]
Nice!

**Pull request merged**